golem/resources/utils/payment.py

```diff
@@ -0,0 +1,173 @@
+
+
+class PayDocumentStatus(str, Enum):
+    ISSUED = "ISSUED"
+    RECEIVED = "RECEIVED"
+    ACCEPTED = "ACCEPTED"
+    REJECTED = "REJECTED"
+    FAILED = "FAILED"
+    SETTLED = "SETTLED"
+    CANCELLED = "CANCELLED"
+
+
+@dataclass
+class PaymentProps:
+    debit_notes_accept_timeout: Optional[int] = None
+    """golem.com.payment.debit-notes.accept-timeout?"""
+    debit_note_interval: Optional[int] = None
+    """golem.com.scheme.payu.debit-note.interval-sec?"""
+    payment_timeout: Optional[int] = None
+    """golem.com.scheme.payu.payment-timeout-sec?"""
+
+    @classmethod
+    def from_properties(cls, properties: Dict) -> "PaymentProps":
+        return cls(
+            debit_notes_accept_timeout=properties.get(defaults.DEBIT_NOTES_ACCEPT_TIMEOUT),
+            debit_note_interval=properties.get(defaults.DEBIT_NOTES_INTERVAL),
+            payment_timeout=properties.get(defaults.PAYMENT_TIMEOUT),
+        )
+
+
+def eth_decimal(value: Union[str, Decimal]) -> Decimal:
+    return Decimal(value).quantize(ETH_SCALE, ROUND_CEILING)
+
+
+def validate_payment_max_cost(
+    coeffs: LinearCoeffs,
+    inf: InfrastructureProps,
+    duration: timedelta,
+    amount: Decimal,
+    time_since_last_debit_note: Optional[timedelta] = None,
+    amount_since_last_debit_note: Optional[Decimal] = None,
+):
+    """Validate payment data max cost.
+
+    Raises: PaymentValidationException
+    """
+    max_cost = eth_decimal(
+        coeffs.price_storage_gib * Decimal(inf.storage_gib)
+        + coeffs.price_mem_gib * Decimal(inf.memory_gib)
+        + coeffs.price_cpu_sec * Decimal(inf.cpu_threads) * Decimal(duration.total_seconds())
+        + coeffs.price_duration_sec * Decimal(duration.total_seconds())
+        + coeffs.price_initial
+    )
+
+    if amount > max_cost:
+        raise PaymentValidationException(
+            "Total amount due exceeds expected max possible cost " f"{amount} > {max_cost}"
+        )
+
+    if time_since_last_debit_note is None or amount_since_last_debit_note is None:
+        return max_cost, None
+
+    max_cost_since_last_debit_note = eth_decimal(
+        coeffs.price_storage_gib * Decimal(inf.storage_gib)
+        + coeffs.price_mem_gib * Decimal(inf.memory_gib)
+        + coeffs.price_cpu_sec
+        * Decimal(inf.cpu_threads)
+        * Decimal(time_since_last_debit_note.total_seconds())
+        + coeffs.price_duration_sec * Decimal(time_since_last_debit_note.total_seconds())
+        + coeffs.price_initial
+    )
+
+    if amount_since_last_debit_note > max_cost_since_last_debit_note:
+        raise PaymentValidationException(
+            "Amount due since last debit note exceeds expected max possible cost "
+            f"{amount_since_last_debit_note} > {max_cost_since_last_debit_note}"
+        )
+
+    return max_cost, max_cost_since_last_debit_note
+
+
+def validate_payment_calculated_cost(
+    coeffs: LinearCoeffs,
+    amount: Decimal,
+    usage_counter_vector: List,
+):
```

Comment:
same here, it's not obvious that a function that peforms `validation` actually returns a calculated value and not e.g. a boolean.

Reply:
added return type and pydoc